<a href="https://www.kaggle.com/code/ehetshamshaukat/cyber-security-salary?scriptVersionId=103336240" target="_blank"><img align="left" alt="Kaggle" title="Open in Kaggle" src="https://kaggle.com/static/images/open-in-kaggle.svg"></a>

## Analysis

## import libraries

In [1]:
import numpy as np
import pandas as pd
import plotly
from plotly import offline
from plotly import graph_objs as go 

## reading data 

In [2]:
dataset=pd.read_csv("../input/cyber-security-salaries/salaries_cyber.csv")
dataset.sample(5)

work_year experience_level employment_type                  job_title  \
413       2022               EN              FT     Cyber Security Analyst   
850       2021               EN              FT             Ethical Hacker   
435       2022               MI              FT  Incident Response Analyst   
858       2021               MI              FT     Cyber Security Analyst   
213       2022               SE              FT           Security Analyst   

     salary salary_currency  salary_in_usd employee_residence  remote_ratio  \
413   55000             USD          55000                 US             0   
850  119000             USD         119000                 US           100   
435   87000             USD          87000                 US            50   
858  150000             USD         150000                 US            50   
213  109280             USD         109280                 US           100   

    company_location company_size  
413               US            M  
850               US            L  
435               US            L  
858               US            M  
213               US            M

## by work year

In [3]:
dataset["work_year"].unique()

array([2022, 2021, 2020])

In [4]:
dataset["work_year"].value_counts(sort=True)

2022    579
2021    484
2020    184
Name: work_year, dtype: int64

In [5]:
dataset.groupby(["work_year","employment_type"]).agg(experience_level=("experience_level","count")).reset_index()

work_year employment_type  experience_level
0       2020              CT                 3
1       2020              FT               179
2       2020              PT                 2
3       2021              CT                 2
4       2021              FL                 3
5       2021              FT               475
6       2021              PT                 4
7       2022              CT                 6
8       2022              FT               571
9       2022              PT                 2

## by experience level

In [6]:
dataset["experience_level"].unique()

array(['EN', 'MI', 'EX', 'SE'], dtype=object)

In [7]:
dataset["experience_level"].value_counts(sort=True)

SE    547
MI    412
EN    215
EX     73
Name: experience_level, dtype: int64

In [8]:
dataset.groupby(["work_year","experience_level"]).agg(salary_by_experience=("salary_in_usd","mean"),salary_by_experience_max=("salary_in_usd","max"),salary_by_experience_min=("salary_in_usd","min")).reset_index()

work_year experience_level  salary_by_experience  \
0        2020               EN          57538.193548   
1        2020               EX         238966.857143   
2        2020               MI          99386.637681   
3        2020               SE         126701.103896   
4        2021               EN          64864.633929   
5        2021               EX         190844.357143   
6        2021               MI          91925.771930   
7        2021               SE         129844.612717   
8        2022               EN          64181.444444   
9        2022               EX         200924.578947   
10       2022               MI         116362.627907   
11       2022               SE         157762.114478   

    salary_by_experience_max  salary_by_experience_min  
0                     122000                      3238  
1                     600000                    117810  
2                     456621                      5337  
3                     300000                      2764  
4                     200000                      2000  
5                     400000                     50000  
6                     320000                      2056  
7                     450000                     35000  
8                     150000                      2000  
9                     360000                     15397  
10                    410000                      2202  
11                    910991                     19568

In [9]:
dataset.groupby("experience_level").agg(average_salaries_usd=("salary_in_usd","mean"),maximim_salaries_usd=("salary_in_usd","max"),minimum_salaries_usd=("salary_in_usd","min")).reset_index()

experience_level  average_salaries_usd  maximim_salaries_usd  \
0               EN          63579.474419                200000   
1               EX         200706.082192                600000   
2               MI         103377.080097                456621   
3               SE         144560.239488                910991   

   minimum_salaries_usd  
0                  2000  
1                 15397  
2                  2056  
3                  2764

## by employment type

In [10]:
dataset["employment_type"].unique()

array(['FT', 'CT', 'PT', 'FL'], dtype=object)

In [11]:
employment_type=dataset["employment_type"].value_counts(sort=True)

In [12]:
labels=employment_type.index
values=employment_type.values
trace0=go.Pie(values=values,labels=labels,hole=0.4,textposition="inside",textinfo="label+percent")
data=[trace0]
layout=dict(title="Employement Type Percentage ",title_x=0.5,annotations=[dict(text='Employment type', x=0.5, y=0.5, font_size=12, showarrow=False)])
fig=dict(data=data,layout=layout)
offline.iplot(fig)

## job title

In [13]:
dataset["job_title"].unique()

array(['Cyber Program Manager', 'Security Analyst', 'IT Security Analyst',
       'Cyber Security Analyst', 'Application Security Architect',
       'Security Researcher', 'Information Security Compliance Analyst',
       'Information Security Compliance Manager', 'Security Specialist',
       'Application Security Specialist', 'Cyber Security Consultant',
       'Head of Information Security',
       'Security Incident Response Engineer', 'Ethical Hacker',
       'Chief Information Security Officer', 'DevSecOps Engineer',
       'IT Security Manager', 'Application Security Engineer',
       'Vulnerability Analyst', 'Cyber Security Engineer',
       'Information Security Analyst',
       'Principal Application Security Engineer',
       'Cyber Security Architect', 'Information Security Manager',
       'SOC Analyst', 'Penetration Tester',
       'Cyber Threat Intelligence Analyst', 'Cloud Security Engineer',
       'Information Security Officer', 'Security Consultant',
       'Informat

In [14]:
dataset["job_title"].nunique()

87

In [15]:
dataset["job_title"].value_counts(sort=True)

Security Engineer                          194
Cyber Security Analyst                     133
Penetration Tester                          70
Information Security Manager                64
Cyber Security Engineer                     63
                                          ... 
Principal Application Security Engineer      1
Application Security Specialist              1
Information Security Compliance Analyst      1
Application Security Architect               1
Concierge Security Engineer                  1
Name: job_title, Length: 87, dtype: int64

## by salary currency

In [16]:
dataset["salary_currency"].unique()

array(['USD', 'BRL', 'GBP', 'EUR', 'INR', 'CAD', 'CHF', 'DKK', 'SGD',
       'AUD', 'SEK', 'MXN', 'ILS', 'PLN', 'NOK', 'IDR', 'NZD', 'HUF',
       'ZAR', 'TWD', 'RUB'], dtype=object)

In [17]:
dataset["salary_currency"].nunique()

21

In [18]:
currency=dataset["salary_currency"].value_counts(sort=True)
currency

USD    934
EUR    127
GBP     56
CAD     39
INR     23
AUD     18
BRL     12
CHF      9
NZD      5
PLN      4
DKK      4
ZAR      3
SEK      3
ILS      2
SGD      2
NOK      1
IDR      1
MXN      1
HUF      1
TWD      1
RUB      1
Name: salary_currency, dtype: int64

In [19]:
labels=currency.index
values=currency.values
trace0=go.Pie(labels=labels,values=values,hole=0.4,textinfo="label+percent+value",textposition="inside")
data=[trace0]
layout=dict(title="Companies paying in different currencies.",title_x=0.5,annotations=[dict(text="Currencies",x=0.5,y=0.5,font_size=14,showarrow=False)])
fig=dict(data=data,layout=layout)
offline.iplot(fig)

by employee residence 

In [20]:
dataset["employee_residence"].unique()

array(['US', 'BR', 'BW', 'GB', 'TW', 'NL', 'IN', 'DE', 'AE', 'CA', 'CH',
       'DK', 'CL', 'ES', 'SG', 'AU', 'FR', 'IT', 'PK', 'GR', 'GH', 'AZ',
       'RO', 'AR', 'DZ', 'AF', 'SI', 'HR', 'SE', 'HU', 'ET', 'MX', 'IL',
       'IE', 'PL', 'CR', 'JP', 'PT', 'NG', 'NO', 'CZ', 'ID', 'EE', 'KG',
       'KE', 'BG', 'IR', 'NZ', 'BE', 'ZA', 'AT', 'LU', 'EG', 'VN', 'SA',
       'LT', 'RU', 'TR'], dtype=object)

In [21]:
dataset["employee_residence"].nunique()

58

## by remote ratio

In [22]:
dataset["remote_ratio"].unique()

array([ 50,   0, 100])

In [23]:
dataset["remote_ratio"].nunique()

3

In [24]:
remote_ratio=dataset["remote_ratio"].value_counts(sort=True)

In [25]:
labels=remote_ratio.index
values=remote_ratio.values
trace0=go.Pie(labels=labels,values=values,hole=0.4,textposition="inside",textinfo="label+percent+value")
data=[trace0]
layout=dict(title="Remote Work Ratio.",title_x=0.5,annotations=[dict(text="Remote ratio",font_size=14,x=0.5,y=0.5,showarrow=False)])
fig=dict(data=data,layout=layout)
offline.iplot(fig)

## by company location

In [26]:
dataset["company_location"].unique()

array(['US', 'BR', 'BW', 'GB', 'SG', 'NL', 'IN', 'DE', 'AE', 'CA', 'CH',
       'DK', 'CL', 'ES', 'AU', 'FR', 'IT', 'JP', 'GR', 'AZ', 'RO', 'DZ',
       'AQ', 'AX', 'SI', 'HR', 'SE', 'HU', 'ET', 'MX', 'IL', 'IE', 'PK',
       'NO', 'PL', 'PT', 'RU', 'CZ', 'ID', 'EE', 'KE', 'RS', 'AR', 'NZ',
       'BE', 'ZA', 'AT', 'UM', 'LU', 'EG', 'TW', 'VN', 'SA', 'AF', 'TR'],
      dtype=object)

In [27]:
dataset["company_location"].nunique()

55

In [28]:
company_location=dataset["company_location"].value_counts(sort=True)

In [29]:
values=company_location.values
labels=company_location.index
trace0=go.Pie(labels=labels,values=values,hole=0.4,textposition="inside",textinfo="label+percent+value")
data=[trace0]
layout=dict(title="Company location distribution",title_x=0.5,annotations=[dict(text="company location",x=0.5,y=0.5,showarrow=False,font_size=14)])
fig=dict(layout=layout,data=data)
offline.iplot(fig)

## by company size

In [30]:
dataset["company_size"].unique()

array(['S', 'M', 'L'], dtype=object)

In [31]:
dataset["company_size"].nunique()

3

In [32]:
company_size=dataset["company_size"].value_counts(sort=True)

In [33]:
labels=company_size.index
values=company_size.values
trace0=go.Pie(values=values,labels=labels,hole=0.4,textposition="inside",textinfo="label+percent")
data=[trace0]
layout=dict(title="Company Size.",title_x=0.5)
fig=dict(data=data,layout=layout)
offline.iplot(fig)

## Answer get by doing analysis
1. Most of companies paying in USD
2. Most of companies are Large size
3. most of employment type is FT
4. most of job title are Security Engineer(194),Cyber Security Analyst(133),Penetration Tester(70)
5. most of company location is in US
6. over 769 companies are offering 100% remote ratio